jax/_src/interpreters/partial_eval.py

```diff
@@ -1971,18 +1981,23 @@ def process_custom_transpose(self, prim, call, tracers, *,
 
 custom_staging_rules: Dict[Primitive, Callable] = {}
 
-def _memoize(thunk):
-  cell = []
+@lu.transformation
+def _interleave_fun(every_others, *args, **kwargs):
+  args_ = [x for pair in zip(args, every_others) for x in pair]
+  yield (yield (args_, kwargs))
+
+def _memoize(fn):
+  cells = {}
   saved_state = [core.thread_local_state.trace_state.copy()]
-  def memoized():
-    if not cell:
+  def memoized(*args):
+    if args not in cells:
       prev_state = core.thread_local_state.trace_state
       core.thread_local_state.trace_state = saved_state.pop()
```

Comment:
Just noticed: I think we can't pop here if we might reuse the same saved_state for different values of `args`.

Reply:
Great catch. I was just scratching my head over this!